### PR TITLE
Normalize generated container class name

### PR DESCRIPTION
**Pull request by alexander-schranz**
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes -
| Related issues/PRs | https://github.com/sulu/sulu/pull/5180, https://github.com/sulu/skeleton/pull/45, https://github.com/sulu/sulu/pull/5189
| License | MIT
| Documentation PR | sulu/sulu-docs#prnum

#### What's in this PR?

Fix generated container class name.

#### Why?

In all contexts admin, website and preview the `getContainerClassName` should be the same as it is used by symfony as cache seed https://github.com/symfony/symfony/blob/v4.4.7/src/Symfony/Component/Cache/DependencyInjection/CachePoolPass.php#L56

### CODE REVIEW

File: src/Sulu/Component/HttpKernel/SuluKernel.php [hunk 1]
```diff
@@ -85,6 +85,41 @@ protected function configureContainer(ContainerBuilder $container, LoaderInterfa
...
+    {
+        return $this->generateContainerClass(static::class);
+    }
+
+    /**
+     * @internal
+     *
+     * This is only for internal use. To get the container class use `getContainerClass` instead.
+     *
+     * This is a copy of the symfony 5.0 getContainerClass which does not include $this->name.
```
Comment (danrot) [inline src/Sulu/Component/HttpKernel/SuluKernel.php:104]: And where does this function include `$this->name`?

Comment (alexander-schranz) [reply src/Sulu/Component/HttpKernel/SuluKernel.php:104]: > which does not include $this->name

The function from symfony 4.4 did include it this->name https://github.com/symfony/symfony/blob/0889df1e88559968eb291c938d25b96e252f7bed/src/Symfony/Component/HttpKernel/Kernel.php#L478

Not sure how to make this clear 🙈 

The function from symfony 5.0 does not longer include that name and thats why its copied so we have the behaviour from symfony 5 also for 4.4.

**Comment by danrot** [comment]
How can I test that? Should the nodes cache work with that change?

File: src/Sulu/Component/HttpKernel/SuluKernel.php [hunk 2]
```diff
@@ -85,6 +85,41 @@ protected function configureContainer(ContainerBuilder $container, LoaderInterfa
...
+     */
+    protected function getContainerClass()
+    {
+        return $this->generateContainerClass(static::class);
+    }
+
+    /**
+     * @internal
+     *
+     * This is only for internal use. To get the container class use `getContainerClass` instead.
```
Comment (danrot) [inline src/Sulu/Component/HttpKernel/SuluKernel.php:102]: `This is only used to support Symfony 4.4 and 5 at the same time.` - How about something like that?

File: src/Sulu/Component/HttpKernel/SuluKernel.php (hunk 1 above)
Comment (danrot) [reply src/Sulu/Component/HttpKernel/SuluKernel.php:104]: Ok, I've misread :see_no_evil:

**Comment by alexander-schranz** [comment]
@danrot yes the node cache should then work and invalidate correcty.

See https://github.com/sulu/skeleton/pull/45/files for configuration.

File: src/Sulu/Bundle/PreviewBundle/Preview/Renderer/PreviewKernel.php [hunk 3]
```diff
@@ -43,6 +43,18 @@ public function registerContainerConfiguration(LoaderInterface $loader)
         $loader->load(implode(DIRECTORY_SEPARATOR, [__DIR__, '..', '..', 'Resources', 'config', 'config_preview.yml']));
     }
 
+    /**
+     * The "getContainerClass" need to be normalized for preview and over contexts
```
Comment (danrot) [inline src/Sulu/Bundle/PreviewBundle/Preview/Renderer/PreviewKernel.php:47]: Should `over` be `other`? Reads strange :thinking: Same in the other comments.

**Comment by danrot** [comment]
Shouldn't that PR also include setting the node cache, so that I have something to test? Or have you done that in a separate PR?

**Comment by danrot** [comment]
But shouldn't that be included in this PR as well? We've also got a production config in sulu/sulu, don't we?

**Pull request merged**